Fetch all products API v2.0

kamal0808 opened the pull request:
Covers fetch all products with API v2.0, and respective tests.
Does not cover any other APIs as I intend to them in their own feature branches (baby-steps)

kamal0808 committed:
Fetch all products API v2.0

pulkitsinghal commented on vend.js:
```diff
@@ -1449,12 +1483,77 @@ var fetchAllProducts = function(args, connectionInfo, processPagedResults) {
+};
+
+
+var fetchPaginationInfo = function (args, connectionInfo) {
```
@kamal0808 - suggestions:

1. don't include fetchPaginationInfo in this PR
1. submit a new PR with a test for the old version
1. submit a new PR with the changes to fetchPaginationInfo
    1. either the old test will keep working
    1. or fetchPaginationInfo will have to be tweaked for backward compatibility
    1. or fetchPaginationInfo will have breaking changes BUT the updated test will serve as a clue for developers on how to update their own code

pulkitsinghal replied:
Also add `args` to `fetchAllProducts2` so that we don't suffer later when a client cannot handle all the entries at the same time and wants them aggregated using a custom page size like 5k or 10k

pulkitsinghal replied:
so after giving it some thought, i've come to the conclusion that if the developer providing the `args`, does not set a `pageSize` then we should not try to set any defaults

pulkitsinghal replied:
Please review PR #17 then merge it to develop then rebase this PR and cut/paste your code into `product.js` file

kamal0808 commented:
@pulkitsinghal says: lets add `args` to `fetchAllProducts2` so that we don't suffer later when a client cannot handle all the entries at the same time

Commit:
fixes based on code review discussion

1. add args to fetchAllProducts2 so that we don't suffer later
   when a client cannot handle all the entries at the same time
   and wants them aggregated using a custom page size

2. if the developer providing the args, does not set a pageSize
   then we should not try to set any default pageSize as the user's
   process may or may not be able to handle the massive size that we
   deem appropriate as a default (40k?) ... let's leave it at
   whatever default vend chooses

pulkitsinghal commented:
@kamal0808 - i've rebased and updated your PR with my previous suggestions ... feel free to merge or ask questions or refine even further

kamal0808 commented:
@pulkitsinghal Some tests still failing, will check those
